Lib/test/test_timeit.py

```diff
@@ -376,15 +397,39 @@ def callback(a, b):
             num_loops, time_taken = self.autorange(callback=callback)
         self.assertEqual(num_loops, 500)
         self.assertEqual(time_taken, 500/1024)
-        expected = ('1 0.001\n'
-                    '2 0.002\n'
-                    '5 0.005\n'
-                    '10 0.010\n'
-                    '20 0.020\n'
-                    '50 0.049\n'
-                    '100 0.098\n'
-                    '200 0.195\n'
-                    '500 0.488\n')
+        expected = dedent('''\
+                    1 0.001
+                    2 0.002
+                    5 0.005
+                    10 0.010
+                    20 0.020
+                    50 0.049
+                    100 0.098
+                    200 0.195
+                    500 0.488
+        ''')
+        self.assertEqual(s.getvalue(), expected)
+
+    def test_autorange_with_callback_and_target_time(self):
+        def callback(a, b):
+            print("{} {:.3f}".format(a, b))
+        with captured_stdout() as s:
+            num_loops, time_taken = self.autorange(callback=callback,
+                                                    target_time=0.6)
+        self.assertEqual(num_loops, 1000)
+        self.assertEqual(time_taken, 1000/1024)
+        expected = dedent('''\
+                    1 0.001
+                    2 0.002
+                    5 0.005
+                    10 0.010
+                    20 0.020
+                    50 0.049
+                    100 0.098
+                    200 0.195
+                    500 0.488
+                    1000 0.977
+        ''')
```

Comment:
I think it would be better to factor out common code instead of duplicating the previous test method.

Reply:
Ok, I'm going to delete that test.